src/opm/input/eclipse/Schedule/RFTConfig.cpp

```diff
@@ -21,196 +21,281 @@
+        && (well_iter->second != RFT::FOPN);
+}
+
+bool RFTConfig::plt() const
+{
+    return ! this->plt_state.empty();
 }
 
-bool RFTConfig::plt(const std::string& wname) const {
-    auto well_iter = this->plt_state.find(wname);
-    if (well_iter == this->plt_state.end())
-        return false;
-    return true;
+bool RFTConfig::plt(const std::string& wname) const
+{
+    return this->plt_state.find(wname) != this->plt_state.end();
 }
 
-std::optional<RFTConfig> RFTConfig::next() const {
-    auto yes_iter_rft = std::find_if( this->rft_state.begin(), this->rft_state.end(), [](const auto& rft_pair) { return rft_pair.second == RFT::YES; });
-    auto yes_iter_plt = std::find_if( this->plt_state.begin(), this->plt_state.end(), [](const auto& plt_pair) { return plt_pair.second == PLT::YES; });
-    if (yes_iter_rft == this->rft_state.end() && yes_iter_plt == this->plt_state.end())
+std::optional<RFTConfig>
+RFTConfig::well_open(const std::string& wname) const
+{
+    auto iter = this->open_wells.find(wname);
+    if (iter != this->open_wells.end()) {
+        // RFT data at well open is already recorded.  Don't trigger new RFT
+        // output event.
         return {};
+    }
 
     auto new_rft = *this;
-    for (auto iter = new_rft.rft_state.begin(); iter != new_rft.rft_state.end(); ) {
-        if (iter->second == RFT::YES)
-            iter = new_rft.rft_state.erase(iter);
-        else
-            iter++;
+
+    if (this->first_open_rft) {
+        // Well opens at this time and user requests RFT data on well open
+        // for all new wells.  Trigger RFT output.
+        new_rft.update(wname, RFT::YES);
+        new_rft.open_wells.insert_or_assign(wname, true);
+
+        return new_rft;
     }
 
-    for (auto iter = new_rft.plt_state.begin(); iter != new_rft.plt_state.end(); ) {
-        if (iter->second == PLT::YES)
-            iter = new_rft.plt_state.erase(iter);
-        else
-            iter++;
+    auto rft_fopn = new_rft.rft_state.find(wname);
+    if ((rft_fopn != new_rft.rft_state.end()) &&
+        (rft_fopn->second == RFT::FOPN))
+    {
+        // Well opens at this time and user requests RFT data on well open
+        // for this particular well.  Trigger RFT output.
+        rft_fopn->second = RFT::YES;
+    }
+
+    new_rft.open_wells.insert_or_assign(wname, true);
+
+    return new_rft;
+}
+
+std::optional<RFTConfig> RFTConfig::next() const
+{
+    // Next block configured by removing all 'YES' nodes from *this.  We do
+    // this because the 'YES' nodes have already triggered by the time the
+    // next block runs.
+    const auto rft_is_yes = [](const auto& rft_pair) {
+        return rft_pair.second == RFT::YES;
+    };
+
+    const auto plt_is_yes = [](const auto& plt_pair) {
+        return plt_pair.second == PLT::YES;
+    };
+
+    const auto rft_has_yes = mapContains(this->rft_state, rft_is_yes);
+    const auto plt_has_yes = mapContains(this->plt_state, plt_is_yes);
```

Comment:
Somehow this looks expensive. Could we do this in one pass returning two values?

Reply:
> Could we do this in one pass returning two values?

Not really, no.  I appreciate that it's subtle, but the first call here inspects the `rft` state while the second call inspects the `plt` state.  Those are two different collections of (slightly) different types.  There is also a third data type, which we currently don't support but anticipate to support in the near future, that deals with segment-related data.